File: cs/synciv.cs
public void PlayerDied()
{
    PlayerDeathData deathData = new PlayerDeathData();
    deathData.PlayerId = this.PlayerId;
    deathData.DeathTime = DateTime.Now;

    NetworkManager.SendDeathData(deathData);
}

public void HandleDeathData(PlayerDeathData deathData)
{
    NetworkManager.BroadcastDeathData(deathData);
}

public void HandleRemoteDeathData(PlayerDeathData deathData)
{
    Player remotePlayer = PlayerList.Find(p => p.PlayerId == deathData.PlayerId);
    if (remotePlayer != null)
    {
        remotePlayer.Die(deathData.DeathTime);
    }
}

File: cpp/synciv.cpp
void playerDied(int playerId) {
    PlayerDeathData deathData;
    deathData.playerId = playerId;
    deathData.deathTime = std::chrono::system_clock::now();

    NetworkManager::sendDeathData(deathData);
}

void handleDeathData(PlayerDeathData deathData) {
    NetworkManager::broadcastDeathData(deathData);
}

void handleRemoteDeathData(PlayerDeathData deathData) {
    for (auto& remotePlayer : PlayerList) {
        if (remotePlayer.getPlayerId() == deathData.playerId) {
            remotePlayer.die(deathData.deathTime);
        }
    }
}